spec/javascript/components/atoms/Button.test.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react'
+import { shallow } from 'enzyme'
+import Button from '~/components/atoms/Button'
+
+describe('Button', () => {
+  test('it renders a button with the correct default properties', () => {
+    const wrapper = shallow(<Button text='This is a button' />)
+    const selector = 'button'
+    expect(wrapper.find(selector).exists()).toBeTruthy()
+    expect(wrapper.find({ type: 'button' })).toHaveLength(1)
```

Comment:
@james-wills-sf @akegan i wanted to use `toHaveProperty` instead of the object property selector for `find` but was unsuccessful.  tried
```
expect(wrapper.find(selector)).toHaveProperty('type', 'button')
```

Reply:
@aekong so here's a general rule to use when writing expects:

1. Anything chained on a wrapper is enzyme + react specific
2. Anything chained after expect() is jest _and therefore not react specific_
   - Won't know about props, state, how to work with wrapper objects

So in your example:
```
    expect(wrapper.find(selector)).toHaveProperty('type', 'button')
```
**wrapper.find(selector)** is the react/enzyme specific piece
**.toHaveProperty('type', 'button')** is the jest piece

So `toHaveProperty` isn't checking for react props, it's probably just looking for regular JSObject properties.

The best way to check for props is like this:
```
expect(wrapper.find(selector).props().type).toEqual('button')
```

Breaking this line down:
**wrapper.find(selector).props().type** is all enzyme/react stuff that resolves to a simple string
 **toEqual('button')** is jest stuff that's just comparing two strings!

Reply:
I should clarify that some jest stuff works fine with wrappers, mostly because enzyme and react are pretty good about using regular javascript arrays and objects where possible

Like `.toHaveLength(x)` is a good method of  checking to see if a wrapper exists, because a single wrapper has a length property = 1, and `wrapper.find()` will return a list of wrappers if multiple are found, which obviously also has a length property

Reply:
super helpful, thanks!